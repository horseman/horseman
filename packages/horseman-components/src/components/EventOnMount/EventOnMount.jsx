import React from "react";
import PropTypes from "prop-types";

class EventOnMount extends React.Component {
  componentDidMount() {
    const { onMount } = this.props;

    onMount();
  }

  componentDidUpdate() {
    const { onUpdate } = this.props;

    onUpdate();
  }

  render() {
    return <div>{this.props.children}</div>;
  }
}

EventOnMount.propTypes = {
  onMount: PropTypes.func,
  onUpdate: PropTypes.func,
  children: PropTypes.node,
};

export default EventOnMount;
