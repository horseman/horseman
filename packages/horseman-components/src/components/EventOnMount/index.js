import EventOnMount from "./EventOnMount";

export default EventOnMount;
