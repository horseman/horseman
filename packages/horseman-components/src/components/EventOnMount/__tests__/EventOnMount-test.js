import React from "react";
import { shallow } from "enzyme";

import EventOnMount from "../";

const TestComponent = () => <div />;

describe("EventOnMount", () => {
  test("Fires Component Did Mount only", () => {
    const spy = jest.spyOn(EventOnMount.prototype, "componentDidMount");
    const gtmSpy = jest.spyOn(EventOnMount.prototype, "fireGtmEvent");
    const wrapper = shallow(
      <EventOnMount>
        <TestComponent />
      </EventOnMount>,
    );
    expect(wrapper).toMatchSnapshot();
    expect(spy).toHaveBeenCalled();
    expect(gtmSpy).toHaveBeenCalledTimes(0);
  });

  test("Fires fireGtmEvent", () => {
    window.dataLayer = {
      push: obj => obj,
    };
    const windowSpy = jest.spyOn(window.dataLayer, "push");
    const gtmSpy = jest.spyOn(EventOnMount.prototype, "fireGtmEvent");
    const wrapper = shallow(
      <EventOnMount gtmEvent="loaded">
        <TestComponent />
      </EventOnMount>,
    );
    expect(wrapper).toMatchSnapshot();
    const event = "loaded";
    expect(gtmSpy).toHaveBeenCalled();
    expect(windowSpy).toHaveBeenCalledWith({ event });
  });
});
