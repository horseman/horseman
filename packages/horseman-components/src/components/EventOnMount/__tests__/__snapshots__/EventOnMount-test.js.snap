// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventOnMount Fires Component Did Mount only 1`] = `
<div>
  <TestComponent />
</div>
`;

exports[`EventOnMount Fires fireGtmEvent 1`] = `
<div>
  <TestComponent />
</div>
`;
